packages/node_modules/@node-red/util/lib/i18n.js

```diff
@@ -50,7 +50,7 @@ async function registerMessageCatalog(namespace,dir,file) {
     return initPromise.then(function() {
         return new Promise((resolve,reject) => {
             resourceMap[namespace] = { basedir:dir, file:file, lngs: []};
-            fs.readdir(dir,function(err, files) {
+            fs.promises.readdir(dir,function(err, files) {
```

Comment:
```suggestion
            fs.readdir(dir,function(err, files) {
```

`fs-extra` can work in both callback and promise mode, depending on whether you provide a callback or not.

In the existing code here, we have a callback - it isn't using the promise version. Changing it to `fs.promises` means `readdir` will be returning a promise rather than call the callback - which is why the tests fail as this never completes.

So for this one, leave it as it was and the tests start passing.

Reply:
Thanks. much cleaner.